grpc-api/src/main/java/io/deephaven/grpc_api/table/ops/AsOfJoinTablesGrpcImpl.java

```diff
@@ -0,0 +1,88 @@
+package io.deephaven.grpc_api.table.ops;
+
+import com.google.common.collect.Lists;
+import com.google.rpc.Code;
+import io.deephaven.base.verify.Assert;
+import io.deephaven.db.exceptions.ExpressionException;
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.tables.live.LiveTableMonitor;
+import io.deephaven.db.tables.select.MatchPair;
+import io.deephaven.db.tables.select.MatchPairFactory;
+import io.deephaven.grpc_api.session.SessionState;
+import io.deephaven.grpc_api.util.GrpcUtil;
+import io.deephaven.proto.backplane.grpc.AsOfJoinTablesRequest;
+import io.deephaven.proto.backplane.grpc.BatchTableRequest;
+import io.deephaven.proto.backplane.grpc.JoinTablesRequest;
+import io.deephaven.util.FunctionalInterfaces;
+import io.grpc.StatusRuntimeException;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class AsOfJoinTablesGrpcImpl extends GrpcTableOperation<AsOfJoinTablesRequest> {
+
+    private final LiveTableMonitor liveTableMonitor;
+
+    private static final MultiDependencyFunction<AsOfJoinTablesRequest> EXTRACT_DEPS =
+            (request) -> Lists.newArrayList(request.getLeftId(), request.getRightId());
+
+    @Inject
+    protected AsOfJoinTablesGrpcImpl(LiveTableMonitor liveTableMonitor) {
+        super(BatchTableRequest.Operation::getAsOfJoin, AsOfJoinTablesRequest::getResultId, EXTRACT_DEPS);
+        this.liveTableMonitor = liveTableMonitor;
+    }
+
+    @Override
+    public void validateRequest(final AsOfJoinTablesRequest request) throws StatusRuntimeException {
```

Comment:
It may be worthwhile to validate the match rule vs aj & raj. See QueryTable.ajInternal. aj can only be LESS_THAN or LESS_THAN_EQUAL; raj can only be GREATER_THAN or GREATER_THAN_EQUAL (I think).

Reply:
Alternatively, we can do away w/ `Type`, and assume that `<` or `<=` is aj, and `>` or `>=` is raj.

Reply:
That's irritating - why provide one enum when at most half the values ever make sense? Esp in java, where each enum could could be referenced only from its own method...

Reply:
What is your preference here for handling this? Remove "type" and treat all of these as "as of join" just with different ordering? I'm not very clear on the semantics of as of join at all.

Reply:
I'm not sure if we should prefer to "fix" / reduce the duplication at the wire level, or try to match the bad api in java. All else equal, I'd prefer to remove as_of_join_type and infer via MatchRule. @rcaudy ?

Reply:
Patch updated to try this, let me know what you think (conv ongoing in slack too).